src/controllers/expense.controller.js

```diff
@@ -0,0 +1,102 @@
+const expenseService = require('../services/expense.service');
+const userService = require('./../services/user.service');
+
+const getExpenses = async (req, res) => {
+  const { userId, categories, from, to } = req.query;
+
+  try {
+    const expenses = await expenseService.getAllExpense({
+      userId,
+      categories,
+      from,
+      to,
+    });
+
+    res.status(200).send(expenses);
+  } catch (error) {
+    res.status(500).send('Server error');
+  }
+};
+
+const getOneExpense = async (req, res) => {
+  const { id } = req.params;
+
+  try {
+    const expense = await expenseService.getExpenseById(id);
+
+    if (!expense) {
+      return res.status(404).send('Not found');
+    }
+    res.send(expense);
+  } catch (error) {
+    res.status(500).send('Server error');
+  }
+};
+
+const createExpense = async (req, res) => {
+  const data = req.body;
+
+  try {
+    const user = await userService.getUserById(data.userId);
+
+    if (!user) {
+      return res.sendStatus(400);
+    }
+
+    const expense = await expenseService.createExpense(data);
+
+    res.status(201).send(expense);
+  } catch (error) {
+    res.status(500).send('Server error');
+  }
+};
+
+const updateExpense = async (req, res) => {
+  const { id } = req.params;
+  const { title, amount, category, spentAt, note } = req.body;
+
+  try {
+    const expense = await expenseService.getExpenseById(id);
+
+    if (!expense) {
+      return res.status(404).send('Error');
```

Comment:
The error message 'Error' is too generic. Consider using a more descriptive message like 'Expense not found' to provide clarity.